src/test/python/pythonparser/test_pythonparser_3.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) Aniskov N.
+
+import os
+from xml.etree import ElementTree as ET
+from typing import *
+
+import pytest
+
+from src.main.python.pythonparser.pythonparser_3 import *
+from src.main.util.const import RESOURCE_FOLDER_PATH
+
+
+class TestPythonParser3:
+
+    _input_dir = os.path.join(RESOURCE_FOLDER_PATH, 'pythonparser/pythonparser_3')
+
+    test_cases_dirs = \
+        [
+            os.path.join(_input_dir, 'case_small_code_snippets'),
+
+            os.path.join(_input_dir, 'case_big_code_snippets'),
+
+            os.path.join(_input_dir, 'case_classes'),
+
+            os.path.join(_input_dir, 'case_empty_file'),
+
+            pytest.param(os.path.join(_input_dir, 'case_fstrings'),
+                         marks=pytest.mark.skip(reason='fstrings are not supported yet')),
+
+            pytest.param(os.path.join(_input_dir, 'case_python_3_8'),
+                         marks=pytest.mark.skip(reason='python 3.8 not supported yet'))
+        ]
+
+    # Helper functions
+
+    @staticmethod
+    def get_xml_tree(path_to_py_file: str) -> ET.Element:
+        parsed_json = parse_file(path_to_py_file)
+        xml_str = json2xml(parsed_json)
+        root = ET.fromstring(xml_str)
+        return root
+
+    @staticmethod
+    def count_token_occurrence(tree: ET.Element, tokens: Iterable[str]) -> Dict[str, int]:
+        result = {token: len(tree.findall('.//' + token))
+                  for token in tokens}
+        return result
```

Comment:
Please, delete the `result` variable. I think you should use `return {token: len(tree.findall('.//' + token))for token in tokens}`, it look prettier

Reply:
Thanks for noticing, but from my point of view the `result` variable is helpful at least for reasons of debugging. 
What do you think?

Reply:
Why? You can toggle a breakpoint and calculate the necessary result for example. Or get it manually by debugging. What do you think?

Reply:
Yes, maybe you are right. I used to create an intermediate variable with a meaningful name before returning result from function. I mean, for example, here it is clear that root of tree is returned:
```
 root = ET.fromstring(xml_str)
 return root
```
but, of course, the statement above can be simplified to `return  ET.fromstring(xml_str)`
And i wrote
```
result = {token: len(tree.findall('.//' + token))
                  for token in tokens}
return result
```
to make all code more uniform. In the end it's really small thing and if in your experience it's better to do as you say, than i'll do so :)

Reply:
Thank you for the examples! In the first case using an additional variable is a good way, because it is unclear what `ET.fromstring(xml_str)` returns. But in the second example, you can understand easily what is it. I mean it is a good idea to use an additional variable for the difficult cases, but in others better don't use it. I would be happy if you delete it :)